src/firewall/firewalld.rs

```diff
@@ -706,3 +706,51 @@ pub fn rm_firewalld_if_possible(net: &ipnet::IpNet) {
         ),
     };
 }
+
+/// Check whether firewalld's StrictForwardPorts setting is enabled.
+/// Returns false if firewalld is not installed, not running, or there is any
+/// error with the process.
+pub fn is_firewalld_strict_forward_enabled() -> bool {
+    let conn = match Connection::system() {
```

Comment:
looking at firewalld::add_firewalld_if_possible() it might be nice if to share the connecting and not open a new time for every single thing which seems super inefficient for no good reason.

We could call Connection::system() once and is_firewalld_running once as well and then just pass an `Option<Connection>` around.